cmd/reva/transfer-create.go

```diff
@@ -19,42 +19,172 @@
+	ocmprovider "github.com/cs3org/go-cs3apis/cs3/ocm/provider/v1beta1"
 	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
-	tx "github.com/cs3org/go-cs3apis/cs3/tx/v1beta1"
+	ocm "github.com/cs3org/go-cs3apis/cs3/sharing/ocm/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	types "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/pkg/errors"
 )
 
 func transferCreateCommand() *command {
 	cmd := newCommand("transfer-create")
-	cmd.Description = func() string { return "create transfer between 2 remotes" }
-	cmd.Usage = func() string { return "Usage: transfer-create [-flags]" }
+	cmd.Description = func() string { return "create transfer between 2 sites" }
+	cmd.Usage = func() string { return "Usage: transfer-create [-flags] <path>" }
 	grantee := cmd.String("grantee", "", "the grantee, receiver of the transfer")
+	granteeType := cmd.String("granteeType", "user", "the grantee type, one of: user, group")
+	idp := cmd.String("idp", "", "the idp of the grantee, default to same idp as the user triggering the action")
+	// fixed for data transfer
+	rol := editorPermission
 
 	cmd.Action = func(w ...io.Writer) error {
-		// validate flags
+		if cmd.NArg() < 1 {
+			return errors.New("Invalid arguments: " + cmd.Usage())
+		}
+
 		if *grantee == "" {
 			return errors.New("Grantee cannot be empty: use -grantee flag\n" + cmd.Usage())
 		}
+		if *idp == "" {
+			return errors.New("Idp cannot be empty: use -idp flag\n" + cmd.Usage())
+		}
+
+		// the resource to transfer; the path
+		fn := cmd.Args()[0]
 
 		ctx := getAuthContext()
 		client, err := getClient()
 		if err != nil {
 			return err
 		}
 
-		transferRequest := &tx.CreateTransferRequest{}
+		// check if invitation has been accepted
+		acceptedUserRes, err := client.GetAcceptedUser(ctx, &invitepb.GetAcceptedUserRequest{
+			RemoteUserId: &userpb.UserId{OpaqueId: *grantee, Idp: *idp},
+		})
+		if err != nil {
+			return err
+		}
+		if acceptedUserRes.Status.Code != rpc.Code_CODE_OK {
+			return formatError(acceptedUserRes.Status)
+		}
 
-		transferResponse, err := client.CreateTransfer(ctx, transferRequest)
+		// verify resource stats
+		statReq := &provider.StatRequest{
+			Ref: &provider.Reference{
+				Spec: &provider.Reference_Path{
+					Path: fn,
+				},
+			},
+		}
+		statRes, err := client.Stat(ctx, statReq)
 		if err != nil {
 			return err
 		}
-		if transferResponse.Status.Code != rpc.Code_CODE_OK {
-			return formatError(transferResponse.Status)
+		if statRes.Status.Code != rpc.Code_CODE_OK {
+			return formatError(statRes.Status)
 		}
 
+		providerInfoResp, err := client.GetInfoByDomain(ctx, &ocmprovider.GetInfoByDomainRequest{
+			Domain: *idp,
+		})
+		if err != nil {
+			return err
+		}
+
+		_, pint, err := getOCMSharePerm(rol)
+		if err != nil {
+			return err
+		}
+		resourcePermissions := &provider.ResourcePermissions{
+			InitiateFileDownload: true,
+		}
+		datatxProtocol, err := json.Marshal(
```

Comment:
Do we need the json struct? We're not using options anywhere and from the HTTP handler, you don't even pass that. Use
```
"protocol": &types.OpaqueEntry{
	Decoder: "plain",
	Value:   []byte("datatx")
},
```